client/ayon_core/tools/publisher/control.py

```diff
@@ -2389,17 +2549,57 @@ def run_action(self, plugin_id, action_id):
 
         self.emit_card_message("Action finished.")
 
+    #TODO: Test the _is_interactive method in both interactive and batch modes.
+    @property
+    def _is_interactive(self):
+        """Check for Interactive Mode
+
+        Determine if the script is running in interactive mode
+        (as opposed to batch mode) based on the host application.
+        """
+
+        host = os.getenv('AYON_HOST_NAME')
+
+        if host is None:
+            raise EnvironmentError(
+                "Host environment variable 'AYON_HOST_NAME' is not set")
+        elif host == 'maya':
+            import maya.cmds
+            return not maya.cmds.about(query=True, batch=True)
+        elif host == 'houdini':
+            import hou
+            return hou.isUIAvailable()
+        else:
+            raise ValueError(f"Unknown host environment: {host}")
```

Comment:
Let's add Blender for good measure :)
```suggestion
        elif host == 'blender':
            import bpy
            return not bpy.app.background
        else:
            raise ValueError(f"Unknown host environment: {host}")
```

Reply:
added 👍